lib/validator/validator.cpp

```diff
@@ -165,32 +229,49 @@ Expect<void> Validator::validate(const AST::SubType &Type) {
     }
   }
 
-  // In current version, the length of type index vector will be <= 1.
-  if (Type.getSuperTypeIndices().size() > 1) {
-    spdlog::error(ErrCode::Value::InvalidSubType);
-    spdlog::error("    Accepts 1 super type currently."sv);
-    return Unexpect(ErrCode::Value::InvalidSubType);
+  if (!Conf.hasProposal(Proposal::GC)) {
+    // When GC proposal is not enabled, the length of type
+    // index vector must be <= 1.
+    if (Type.getSuperTypeIndices().size() > 1) {
+      spdlog::error(ErrCode::Value::InvalidSubType);
+      spdlog::error(
+          "    Accepts 1 super type currently when GC proposal is disabled."sv);
+      return Unexpect(ErrCode::Value::InvalidSubType);
+    }
   }
-  for (auto Index : Type.getSuperTypeIndices()) {
-    if (Index >= TypeVec.size()) {
+
+  for (const auto &Index : Type.getSuperTypeIndices()) {
+    if (unlikely(Index >= TypeVec.size())) {
       spdlog::error(ErrCode::Value::InvalidSubType);
       spdlog::error(
-          ErrInfo::InfoForbidIndex(ErrInfo::IndexCategory::DefinedType, Index,
-                                   static_cast<uint32_t>(TypeVec.size())));
+          "    Super type not found. Index {} is out of bounds for defined types (size: {})."sv,
+          Index, TypeVec.size());
       return Unexpect(ErrCode::Value::InvalidSubType);
     }
+
+    // Conditional Check: Subtype depth is only validated with GC
+    if (Conf.hasProposal(Proposal::GC)) {
```

Comment:
The logic is completely different between your previous version and the current one. Could you please check again if this is your intention?

Reply:
Actually the proposal checking condition is not necessary.
With the subtypes which have super type indices, the binary format syntax are accept only when the GC proposal turning on.